src/inewsFtp.ts

```diff
@@ -0,0 +1,447 @@
+		} else return undefined
+	}
+
+	/**
+	 * Get the story ID from the fileName (in XXXXXX:YYYYYY:ZZZZZZ, it will return XXXXXX)
+	 * http://resources.avid.com/SupportFiles/attach/Broadcast/inews-ftp-server.pdf
+	 * @param fileName
+	 * @returns {*}
+	 * @private
+	 */
+	private _storyIdentifierFromFilename(fileName: string) {
+		const fileParts = fileName.split(':')
+		return fileParts[0]
+	}
+
+	/**
+	 * Get the story locator from the fileName (in XXXXXX:YYYYYY:ZZZZZZ, it will return YYYYYY:ZZZZZZ)
+	 * http://resources.avid.com/SupportFiles/attach/Broadcast/inews-ftp-server.pdf
+	 * @param fileName
+	 * @returns {*}
+	 * @private
+	 */
+	private _storyLocatorFromFilename(fileName: string) {
+		const fileParts = fileName.split(':')
+		return fileParts[1] + ':' + fileParts[2]
+	}
+
+	private _flagsFromListItem(listItem: string): { floated?: boolean } {
+		const flags: { floated?: boolean } = {}
+		const pattern = /([^\s]+)/i
+		const flagParts = listItem.match(pattern)
+
+		if (flagParts) {
+			flags.floated = flagParts[0][1] == 'f'
+		}
+
+		return flags
+	}
+
+	private _dateFromListItem(listItem: string) {
+		const pattern = / ([A-Za-z]{3,4})[ ]+([0-9]{1,2})[ ]+([0-9]{4}|([0-9]{1,2}):([0-9]{2}))/i
+		const dateParts = listItem.match(pattern)
+
+		if (!dateParts) {
+			return undefined
+		}
+
+		try {
+			if (typeof dateParts[4] !== 'undefined') {
+				const dateNow = new Date()
+				const dateModified = new Date(
+					dateParts[1] + ' ' + dateParts[2] + ' ' + dateNow.getFullYear() + ' ' + dateParts[3]
+				)
+				if (dateModified.getMonth() > dateNow.getMonth())
+					// change to last year if the date would fall in the future
+					dateModified.setFullYear(dateNow.getFullYear() - 1)
+				return dateModified
+			} else return new Date(dateParts[0])
+		} catch (error) {
+			return undefined
+		}
+	}
+
+	private _queueFromListItem(listItem: string) {
+		const pattern = /.([A-Za-z0-9-]*)$/
+		const matchParts = listItem.match(pattern)
+		return matchParts === null ? undefined : matchParts[1]
+	}
+
+	private _fileNameFromListItem(listItem: string) {
+		const pattern = /[A-Z0-9]{8}:[A-Z0-9]{8}:[A-Z0-9]{8}/i
+		const matchParts = listItem.match(pattern)
+		return matchParts === null ? undefined : matchParts[0]
+	}
+
+	private _storyNameFromListItem(listItem: string) {
+		const pattern = /(?:[0-9A-F]{8}:?){3} (.+?)$/
+		const listItemParts = listItem.match(pattern)
+		return Array.isArray(listItemParts) && listItemParts.length > 1 ? listItemParts[1] : ''
+	}
+
+	private _objectMerge(...args: Array<Record<string, unknown>>): Record<string, unknown> {
+		const merged: Record<string, unknown> = {}
+		this._objectForEach(args, (argument: Record<string, unknown>) => {
+			for (const attrname in argument) {
+				if ({}.hasOwnProperty.call(argument, attrname)) {
+					merged[attrname] = argument[attrname]
+				}
+			}
+		})
+		return merged
+	}
+
+	private _objectForEach<T extends Array<Record<string, unknown>> | Record<string, unknown>>(
```

Comment:
Would be nice with consistent use of array type syntax. Can we change this to:

```suggestion
	private _objectForEach<T extends Record<string, unknown>[] | Record<string, unknown>>(
```